src/pipert2/core/handlers/message_handler.py

```diff
@@ -1,5 +1,35 @@
 from abc import ABC, abstractmethod
+from pipert2.core.base.message import Message
 
 
 class MessageHandler(ABC):
-    pass
+    """The message handler is responsible for receiving messages
+    and relaying them onward if needed.
+    The message handler communicates only with the network object
+    and not with other routines directly.
+    """
+
+    def __init__(self, input_obj, output_obj):
+        self.input = input_obj
+        self.output = output_obj
```

Comment:
Not sure if both of thoes data members are needed.
For example in the redis implementation that we had, we only needed one obj which is the redis connection and used it both for the input and output.